### ISSUE TITLE
Feature request: Check that globs in `on.*.paths` match at least one file

### ISSUE THREAD

**Issue by allanlewis** [rhysd/actionlint#430]
It would be useful to me if Actionlint could report when globs in `on.*.paths` don't match any files.

The following shell pipeline approximates what I'd like to do:

```shell
git ls-files -- '.github/workflows/*.yml' \
  | parallel "yq -o json {} | jq -r '.on | to_entries[].value | .paths // empty | .[]' 2>/dev/null" \
  | sort -u \
  | parallel -k --bar --tag "git ls-files --error-unmatch :{} >/dev/null"
```

However, this obviously uses Git's pattern syntax which probably differs from GitHub's in some subtle respects.

**Comment by anttiharju** [comment]
https://github.com/mpalmer/action-validator can do this. In my use it appears to be the only issue it catches that actionlint does not. I would love it if actionlint could catch this issue because I find action-validator cumbersome to use (have to explicitly feed it files to validate, its install process in CI creates temporary files that interfere with other validation checks I do) compared to actionlint.
